src/server/engine_shard_set.cc

```diff
@@ -390,7 +390,7 @@ void EngineShard::PollExecution(const char* context, Transaction* trans) {
     }
     ++stats_.ooo_runs;
 
-    bool keep = trans->RunInShard(this);
+    bool keep = trans->RunInShard(this, true);
     DLOG_IF(INFO, !dbg_id.empty()) << "Eager run " << sid << ", " << dbg_id << ", keep " << keep;
```

Comment:
Maybe I should add there a `DCHECK_EQ(trans->GetTxqPos(shard) != TxQueue::kEnd,  keep)`? Requires adding `GetTxqPos`

Reply:
Added